addons/payment_authorize/models/payment.py

```diff
@@ -276,12 +276,12 @@ def _authorize_s2s_validate(self, tree):
                     'date_validate': fields.Datetime.now(),
                 })
             if tree.get('x_type').lower() == 'auth_only':
-                if self.callback_eval:
-                    safe_eval(self.callback_eval, {'self': self})
                 self.write({
                     'state': 'authorized',
                     'acquirer_reference': tree.get('x_trans_id'),
                 })
+                if self.callback_eval:
+                    safe_eval(self.callback_eval, {'self': self})
```

Comment:
no, because if the tx is in state authorized, it means the callback has been called once already; so you should do sthg like
```python
                 init_state = self.state
                 self.write({
                     'state': 'done',
                     'acquirer_reference': tree.get('x_trans_id'),
                      'date_validate': fields.Datetime.now(),
                  })
                 if self.callback_eval and init_state != 'authorized':
                     safe_eval(self.callback_eval, {'self': self})
```
or something of this kind.